src/System.Runtime/tests/System/DecimalTests.cs

```diff
@@ -1455,6 +1455,159 @@ public static void Test()
             }
         }
 
+        [Fact]
+        public static void BigInteger_Floor()
+        {
+            decimal[] decimalValues = GetRandomData(out BigDecimal[] bigDecimals);
+            for (int i = 0; i < decimalValues.Length; i++)
+            {
+                decimal d1 = decimalValues[i];
+                BigDecimal expected = bigDecimals[i].Floor();
+                decimal actual = decimal.Floor(d1);
+                unsafe
+                {
+                    if (expected.Scale != (byte)(*(uint*)&actual >> BigDecimal.ScaleShift) || expected.CompareTo(new BigDecimal(actual)) != 0)
+                        throw new Xunit.Sdk.AssertActualExpectedException(expected, actual, d1 + " Floor");
+                }
+            }
+        }
+
+        [Fact]
+        public static void BigInteger_Ceiling()
```

Comment:
Any reason these tests are called `BigInteger`?  I guess I would have thought they would be called `BigDecimal`.

Reply:
I used the same name for previous `BigInteger` based tests. I guess they could be called `BigDecimal` based also, but the `BigDecimal` type is a private type inside this test class only.